proposals/extensions.md

```diff
@@ -497,36 +497,6 @@ Whenever extension members are used in source, we will emit those as reference t
 For example: an invocation of `enumerableOfInt.Method()` would be emitted as a static call 
 to `IEnumerableExtensions.Method<int>(enumerableOfInt)`.  
 
-Note: the metadata representation supports static extension methods that differ in return type.
```

Comment:
Why are we removing the note?

Reply:
It's out-of-date. We produce signature conflict errors for those.